### ISSUE TITLE
[bug]: "expiresIn" should be a number of seconds or string representing a timespan

### ISSUE THREAD

**Issue by a224327780** [hoppscotch/hoppscotch#4205]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### env
```
TOKEN_SALT_COMPLEXITY=10
MAGIC_LINK_TOKEN_VALIDITY=3
REFRESH_TOKEN_VALIDITY="604800000"
ACCESS_TOKEN_VALIDITY="86400000"
```

### error
```
Backend Server | [Nest] 58  - 07/23/2024, 5:55:27 AM   ERROR [ExceptionsHandler] "expiresIn" should be a number of seconds or string representing a timespan eg: "1d", "20h", 60
Backend Server | Error: "expiresIn" should be a number of seconds or string representing a timespan eg: "1d", "20h", 60
Backend Server |     at module.exports [as sign] (/usr/src/app/node_modules/.pnpm/jsonwebtoken@9.0.0/node_modules/jsonwebtoken/sign.js:207:22)
Backend Server |     at JwtService.sign (/usr/src/app/node_modules/.pnpm/@nestjs+jwt@10.1.1_@nestjs+common@10.2.7_reflect-metadata@0.1.13_rxjs@7.6.0_/node_modules/@nestjs/jwt/dist/jwt.service.js:35:20)
Backend Server |     at AuthService.generateRefreshToken (/usr/src/app/packages/hoppscotch-backend/dist/auth/auth.service.js:76:52)
Backend Server |     at AuthService.generateAuthTokens (/usr/src/app/packages/hoppscotch-backend/dist/auth/auth.service.js:94:41)
Backend Server |     at AuthController.githubAuthRedirect (/usr/src/app/packages/hoppscotch-backend/dist/auth/auth.controller.js:74:51)
Backend Server |     at /usr/src/app/node_modules/.pnpm/@nestjs+core@10.2.7_@nestjs+common@10.2.7_reflect-metadata@0.1.13_rxjs@7.6.0__@nestjs+platfor_fugnss62jk6fg5tp4mizlu2qtu/node_modules/@nestjs/core/router/router-execution-context.js:38:29
```

### Steps to reproduce

1 install
2 login

### Environment

Production

### Version

Self-hosted

**Comment by balub** [comment]
Hey @a224327780 can you try removing the quotes and trying ?

**Comment by brummetj** [comment]
+1 to this issue. Im pulling from the latest and im seeing a error with your JWT signing with that environment variable.

```
Backend Server | [Nest] 38  - 07/25/2024, 10:57:58 PM   ERROR [ExceptionsHandler] "expiresIn" should be a number of seconds or string representing a timespan eg: "1d", "20h", 60
Backend Server | Error: "expiresIn" should be a number of seconds or string representing a timespan eg: "1d", "20h", 60
Backend Server |     at module.exports [as sign] (/usr/src/app/node_modules/.pnpm/jsonwebtoken@9.0.0/node_modules/jsonwebtoken/sign.js:207:2
```

**Comment by berkeatalay1** [comment]
+1 same with similar .env file without quotes

**Comment by berkeatalay1** [comment]
Without quotes and comments it works as expected

**Comment by balub** [comment]
@a224327780 are you still facing this issue?

**Comment by lwp2070809** [comment]
> @a224327780 are you still facing this issue?

yes, and this is my env file
```
TOKEN_SALT_COMPLEXITY=10
MAGIC_LINK_TOKEN_VALIDITY=3
REFRESH_TOKEN_VALIDITY=604800000
ACCESS_TOKEN_VALIDITY=86400000
```

**Comment by balub** [comment]
@lwp2070809 are you using a regular docker setup to run the containers or anything else ?

**Comment by lwp2070809** [comment]
Thank you for your interest in this issue.

This is my docker run command:
```bash
docker run -d \
--name hoppscotch \
-p 80:80 \
--env-file .env \
-e TZ=Asia/Shanghai \
--log-driver local \
-m 1G \
--restart unless-stopped \
hoppscotch/hoppscotch:2024.7.0
```

This is the full env file, sensitive information has been removed:
```ini
DATABASE_URL=

JWT_SECRET=SECRET
TOKEN_SALT_COMPLEXITY=10
MAGIC_LINK_TOKEN_VALIDITY=30
REFRESH_TOKEN_VALIDITY=604800000
ACCESS_TOKEN_VALIDITY=86400000
SESSION_SECRET=SECRET

ALLOW_SECURE_COOKIES=true

REDIRECT_URL=https://hoppscotch.example.com
WHITELISTED_ORIGINS=https://hoppscotch.example.com
VITE_ALLOWED_AUTH_PROVIDERS=EMAIL

MAILER_SMTP_ENABLE=true
MAILER_USE_CUSTOM_CONFIGS=true
MAILER_ADDRESS_FROM=
MAILER_SMTP_HOST=
MAILER_SMTP_PORT=
MAILER_SMTP_SECURE=true
MAILER_SMTP_USER=
MAILER_SMTP_PASSWORD=
MAILER_TLS_REJECT_UNAUTHORIZED=false

RATE_LIMIT_TTL=60
RATE_LIMIT_MAX=100

# Base URLs
VITE_BASE_URL=
VITE_SHORTCODE_BASE_URL=
VITE_ADMIN_URL=

# Backend URLs
VITE_BACKEND_GQL_URL=
VITE_BACKEND_WS_URL=
VITE_BACKEND_API_URL=

# Terms Of Service And Privacy Policy Links (Optional)
VITE_APP_TOS_LINK=https://docs.hoppscotch.io/support/terms
VITE_APP_PRIVACY_POLICY_LINK=https://docs.hoppscotch.io/support/privacy

# Set to `true` for subpath based access
ENABLE_SUBPATH_BASED_ACCESS=true

```

When I click the magic link, the error log is:
```
Backend Server | Query Complexity: 7
Backend Server | [Nest] 57  - 08/06/2024, 2:17:18 PM   ERROR [ExceptionsHandler] "expiresIn" should be a number of seconds or string representing a timespan eg: "1d", "20h", 60
Backend Server | Error: "expiresIn" should be a number of seconds or string representing a timespan eg: "1d", "20h", 60
Backend Server |     at module.exports [as sign] (/usr/src/app/node_modules/.pnpm/jsonwebtoken@9.0.0/node_modules/jsonwebtoken/sign.js:207:22)
Backend Server |     at JwtService.sign (/usr/src/app/node_modules/.pnpm/@nestjs+jwt@10.1.1_@nestjs+common@10.2.7_class-transformer@0.5.1_class-validator@0.14.1_refle_kzzd3ma2lqspfbsotrr2bvqo6y/node_modules/@nestjs/jwt/dist/jwt.service.js:35:20)
Backend Server |     at AuthService.generateRefreshToken (/usr/src/app/packages/hoppscotch-backend/dist/auth/auth.service.js:76:52)
Backend Server |     at AuthService.generateAuthTokens (/usr/src/app/packages/hoppscotch-backend/dist/auth/auth.service.js:94:41)
Backend Server |     at AuthService.verifyMagicLinkTokens (/usr/src/app/packages/hoppscotch-backend/dist/auth/auth.service.js:197:35)
Backend Server |     at async AuthController.verify (/usr/src/app/packages/hoppscotch-backend/dist/auth/auth.controller.js:54:28)
```

**Comment by yuyuko233** [comment]
any news?

**Comment by balub** [comment]
@lwp2070809 I cannot reproduce the issue on my end, could you try adding a `s` as in seconds to the end of the numbers and trying again? The library supports this

**Comment by lwp2070809** [comment]
@balub 
Thank God, I finally figured out what the problem was. In the official Hoppscotch documentation or the `.env.example` file in the GitHub repo, only these two variables had inline comments:

```
REFRESH_TOKEN_VALIDITY=604800000 # Default validity is 7 days (604800000 ms) in ms
ACCESS_TOKEN_VALIDITY=86400000 # Default validity is 1 day (86400000 ms) in ms
```

However, environment variables can’t use inline comments like this. As soon as I removed the comments, the issue was gone. Although this is common knowledge, I hope the respected maintainers can update the official documentation to use single-line comments instead, to avoid mistakes for those unfamiliar with this. Thanks a lot!
